main.go

```diff
@@ -58,6 +59,22 @@ func main() {
 	)
 
 	log := logrus.WithFields(logrus.Fields{})
+	e := ctrlog.NewExporter(logger, client)
+	logger.AddHook(e)
+	logrus.RegisterExitHandler(e.Wait)
+
+	// Send aks init data and exit if run from temporary provisioned node.
+	if cfg.AksInitData {
+		h := aks.NewInitDataHandler(log, client)
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+		defer cancel()
+		if err := h.Handle(ctx); err != nil {
+			log.Fatalf("sending aks init data: %v", err)
+		}
+		log.Info("aks init data sent, waiting for init data node shutdown")
+		time.Sleep(time.Hour)
```

Comment:
do we really need this?

Reply:
If pod exists it will be scheduled again. Instead we can wait until node is removed.